server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/ObjectStorageBlobsDAO.java

```diff
@@ -86,10 +87,11 @@
         return AwsS3ObjectStorage.daoBuilder(testConfig);
     }
 
-    public Mono<BucketName> createBucket(BucketName name) {
-        return Mono.fromCallable(() -> blobStore.createContainerInLocation(DEFAULT_LOCATION, name.asString()))
-            .filter(created -> created == false)
-            .doOnNext(ignored -> LOGGER.debug("{} already existed", name))
+    @VisibleForTesting
+    public Mono<BucketName> createBucketIfNotExist(BucketName name) {
+        return Mono.fromCallable(() -> blobStore.containerExists(name.asString()))
+            .filter(existed -> !existed)
+            .map(notExisted -> blobStore.createContainerInLocation(DEFAULT_LOCATION, name.asString()))
```

Comment:
Performance should always come with some kind of bench to justify it.

Here I bet you are eager to say "It's now +5000% time faster in my IDE" so please do not refrain from adding that to the commit message.

Reply:
You have **no atomicity guaranty** hence I will ask you to remove the **ifNotExist** from the name as it is really misleading.